### PR TITLE
create product quantity selector block

### PR DESCRIPTION
**Pull request by jgfidelis**
https://buyquantity--storecomponents.myvtex.com/classic-shoes/p?disableUserLand

Screen Shot 2019-04-10 at 4 14 20 PM

example usage of block:

```
"product-details#default": {
    "blocks": [
      "breadcrumb",
      "product-name",
      "product-images",
      "product-price",
      "product-description",
      "product-specifications",
      "buy-button",
      "sku-selector",
      "shipping-simulator",
      "availability-subscriber",
      "share",
      "product-quantity-selector"
    ],
    "props": {
      "displayVertically": true,
      "share": {
        "social": {
          "Facebook": true,
          "WhatsApp": true,
          "Twitter": false
        }
      },
      "price": {
        "labelSellingPrice": null,
        "showListPrice": true,
        "showLabels": true,
        "showInstallments": true,
        "showSavings": true
      },
      "name": {
        "showBrandName": false,
        "showSku": false,
        "showProductReference": false
      }
    }
  },
"product-quantity-selector": {
    "props": {
      "maximumAvailableQuantity": 9999999
    }
  },
```

#### Types of changes

* [ ] Bug fix (a non-breaking change which fixes an issue)
* [x] New feature (a non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/ProductDetails.js [hunk 1]
```diff
@@ -268,9 +267,9 @@ class ProductDetails extends Component {
   }
 
   render() {
+    console.log('teste FATHER PROPS: ', this.props)
```
Comment (jgfidelis) [inline react/ProductDetails.js]: remove

File: react/ProductDetails.js [hunk 2]
```diff
@@ -4,21 +4,20 @@ import {
...
   reject,
   propOr,
-  pathOr,
 } from 'ramda'
 import { injectIntl, intlShape, FormattedMessage } from 'react-intl'
 
 import { ExtensionPoint, withRuntimeContext } from 'vtex.render-runtime'
 import { Container } from 'vtex.store-components'
 
+
```
Comment (jgfidelis) [inline react/ProductDetails.js]: remove

File: react/ProductDetails.js [hunk 3]
```diff
@@ -268,9 +267,9 @@ class ProductDetails extends Component {
   }
 
   render() {
+    console.log('teste FATHER PROPS: ', this.props)
     const {
       productQuery: { product },
-      slug,
```
Comment (jgfidelis) [inline react/ProductDetails.js:273]: not used

File: react/ProductDetails.js [hunk 4]
```diff
@@ -436,6 +435,17 @@ class ProductDetails extends Component {
                   )}
                   {showBuyButton ? (
                     <div className="pv2 dn db-l mt8">
+                      <ExtensionPoint 
+                        id="product-quantity-selector"
+                        selectedQuantity={selectedQuantity}
+                        onChange={value => this.setState({ selectedQuantity: value })}
+                        availableQuantity={availableQuantity}
+                      />
+                      {/* <QuantityStepper
```
Comment (klzns) [inline react/ProductDetails.js]: commented code

**Review by klzns — CHANGES_REQUESTED** [review]
Missing README

**Review by theomoura — APPROVED** [review]

File: react/components/ProductQuantitySelector/index.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  return (
+    <div className={`${styles.quantitySelectorContainer} flex flex-column mb4`}>
+      <div className="mb3 c-muted-2 t-body">
+        <FormattedMessage id="product-details.quantity" />
+      </div>
+      <NumericStepper
+        size="small"
+        value={selectedQuantity}
+        minValue={1}
+        maxValue={availableQuantity ? availableQuantity : undefined}
```
Comment (matheusps) [inline react/components/ProductQuantitySelector/index.js:19]: Just 
```suggestion
        maxValue={availableQuantity}
```
Would not have the same effect ? 🤔

Comment (jgfidelis) [reply react/components/ProductQuantitySelector/index.js:19]: tenho medo se for 0...

Comment (matheusps) [reply react/components/ProductQuantitySelector/index.js:19]: Ah beleza

File: messages/pt.json [hunk 6]
```diff
@@ -16,5 +16,11 @@
   "editor.product-details.highlights.chooseDefaultSpecification": "Especificações personalizadas",
   "editor.product-details.thumbnailsPosition.title": "Posição das miniaturas",
   "editor.product-details.thumbnailsPosition.left": "Esquerda",
-  "editor.product-details.thumbnailsPosition.right": "Direita"
+  "editor.product-details.thumbnailsPosition.right": "Direita",
+  "product-details.quantity": "Quantidade",
+  "product-details.quantity-available": "{availableQuantity} disponíveis",
```
Comment (klzns) [inline messages/pt.json]: What about `1 disponível`?

File: react/components/ProductQuantitySelector/index.js [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+        onChange={useCallback(e => onChange(e.value), [])}
+      />
+      {showAvailable && <div className="mv4 c-muted-2 t-small">
```
Comment (klzns) [inline react/components/ProductQuantitySelector/index.js]: I would add a css handle here

File: messages/en.json [hunk 8]
```diff
@@ -16,5 +16,11 @@
   "editor.product-details.highlights.chooseDefaultSpecification": "Fields Specifications",
   "editor.product-details.thumbnailsPosition.title": "Thumbnails position",
   "editor.product-details.thumbnailsPosition.left": "Left",
-  "editor.product-details.thumbnailsPosition.right": "Right"
+  "editor.product-details.thumbnailsPosition.right": "Right",
+  "product-details.quantity": "Quantity",
+  "product-details.quantity-available": "{availableQuantity} available",
+  "editor.product-quantity-selector.title": "Product Quantity Selector component",
+  "editor.product-quantity-selector.description": "A component that allows user to select the amount of items to add to cart",
+  "editor.product-quantity-selector.maximumAvailableQuantity.title": "Maxiumum Available Quantity",
```
Comment (klzns) [inline messages/en.json]: It's not actually the maximum available quantity I would say it's the `warningQuantityThreshold`?

Comment (jgfidelis) [reply messages/en.json]: looks like a better name indeed

**Review by klzns — APPROVED** [review]

**Pull request merged**